src/FastRoute/src/FastRouteKernel.cpp

```diff
@@ -587,7 +593,8 @@ void FastRouteKernel::initializeNets(bool reroute)
 
   for (const Net& net : _netlist->getNets()) {
     if (net.getNumPins() > 1
-        && !(_clockNetRouting && net.getSignalType() != odb::dbSigType::CLOCK)
+        && !(_onlyClockNets && net.getSignalType() != odb::dbSigType::CLOCK)
+        && !(_onlySignalNets && net.getSignalType() == odb::dbSigType::CLOCK)
```

Comment:
I think this is written in  a really confusing way.
Again, copying sub expressions.
bool is_clock = net.getSignalType() == odb::dbSigType::CLOCK;
((_onlyClockNets && is_clock) || (_onlySignalNets && !is_clock))